File: src/category_strings/q5_is_permutation/IsPermutation.py
from utils.LogUtil import LogUtil;

class IsPermutation:
    # Write your code here 
    #
    # Determine if one string is a permutation of the other. Not case sensitive. 
    # Not case sensitive.  See README for details
    # @param s - any ASCII characters. 
    # @param t - any ASCII characters. 
    # @return - true if one string is a permutation of the other.
    def is_permutation(self, s: str, t: str) -> bool:
            return s == t;

    def test(self):
        # Change these values to manually test your program
        str_input1 = "Avery likes Shaun";
        str_input2 = "shaun likes avery";
        accepted_response = True;
        actual_response = self.is_permutation(str_input1, str_input2);
        LogUtil().log_results_bool("5: Is Permutation", accepted_response, actual_response);

File: src/category_strings/q5_is_permutation/IsPermutation.js
const LogUtil = require('../../utils/LogUtil.js');

/**
 * Determine if one string is a permutation of the other. Not case sensitive. 
 * Not case sensitive.  See README for details
 * @param {string} s - any ASCII characters. 
 * @param {string} t - any ASCII characters. 
 * @return {boolean} if one string is a permutation of the other.
 */
var IsPermutation = function(s, t) {
    // Write your code here
    return s == t;
}

var test = function() {
    // Change these values to manually test your program
    let strInput1 = "Avery likes Shaun";
    let strInput2 = "shaun likes avery";
    let acceptedResponse = true;
    let actualResponse = IsPermutation(strInput1, strInput2);
    new LogUtil().logResultsBool("5: Is Permutation", acceptedResponse, actualResponse);
}

module.exports = test;